frontend/src/metabase-lib/parameters/utils/parameter-source.unit.spec.ts

```diff
@@ -30,6 +31,54 @@ const searchField = metadata.field(SEARCH_FIELD_ID) as Field;
 const searchField2 = metadata.field(SEARCH_FIELD_2_ID) as Field;
 const noValuesField = metadata.field(NO_VALUES_FIELD_ID) as Field;
 
+describe("getQueryType", () => {
+  describe.each(["string", "location"])("parameter type %s", type => {
+    it.each([
+      ["=", "list"],
+      ["!=", "list"],
+      ["contains", "none"],
+      ["does-not-contain", "none"],
+      ["starts-with", "none"],
+      ["ends-with", "none"],
+    ])(
+      "and %s operator should return %s query type by default",
+      (operator, queryType) => {
+        const parameter = createMockUiParameter({
+          type: `${type}/${operator}`,
+        });
+        expect(getQueryType(parameter)).toBe(queryType);
+      },
+    );
+  });
+
+  describe.each(["number"])("parameter type %s", type => {
+    it.each([
+      ["=", "list"],
+      ["!=", "list"],
+      [">", "none"],
+      ["<", "none"],
+      [">=", "none"],
+      ["<=", "none"],
+      ["between", "none"],
+    ])(
+      "and %s operator should return %s query type by default",
+      (operator, queryType) => {
+        const parameter = createMockUiParameter({
+          type: `${type}/${operator}`,
+        });
+        expect(getQueryType(parameter)).toBe(queryType);
+      },
+    );
+  });
+
+  it('should return "none" query type by default for legacy parameter types', () => {
```

Comment:
The name of the test does not match the assertion (`none` vs `list`).

Reply:
Fixed